src/index.js

```diff
@@ -1,22 +1,52 @@
 import Plugin from '@swup/plugin';
 
+/**
+ * Class representing the Swup Matomo Plugin.
+ * @extends Plugin
+ */
 export default class SwupMatomoPlugin extends Plugin {
 	name = 'SwupMatomoPlugin';
 
+	/**
+	 * Add event handlers on mount
+	 */
 	mount() {
-		this.swup.on('contentReplaced', event => {
-			if (typeof window._paq !== 'undefined') {
-				let title = document.title;
-				let url = window.location.pathname + window.location.search;
-
-				_paq.push(['setDocumentTitle', title]);
-				_paq.push(['setCustomUrl', url]);
-				_paq.push(['trackPageView']);
-
-				this.swup.log(`Matomo pageview (url '${url}').`);
-			} else {
-				console.warn('Matomo is not loaded.');
-			}
-		})
+		this.swup.on('contentReplaced', this.onContentReplaced);
+	}
+
+	/**
+	 * Remove event handlers on mount
+	 */
+	unmount() {
+		this.swup.off('contentReplaced', this.onContentReplaced);
+	}
+
+	/**
+	 * Handles 'contentReplaced'
+	 * @param {(boolean|PopStateEvent)} popstate
+	 * @returns {void}
+	 */
+	onContentReplaced = (popstate) => {
+		this.maybeTrackPageView();
+	};
+
+	/**
+	 * Tracks a page view to matomo, if it is installed
+	 */
+	maybeTrackPageView() {
```

Comment:
IMO the `maybe` is not needed.

Reply:
I don't care much enough to discuss this ;) ...changed the function name to `trackPageView`